api/src/main/java/org/openmrs/module/pihcore/mpi/ImportPatientFromWebService.java

```diff
@@ -0,0 +1,328 @@
+                    }
+                }
+            }
+        }
+        return localPatient;
+    }
+
+    private WebResource setUpWebResource(String serverName, RemoteServerConfiguration remoteServerConfiguration, Integer timeout) {
+        if (remoteServerConfiguration == null) {
+            throw new IllegalArgumentException("Unknown remote server: " + serverName + ". Known servers are " + remoteServers.keySet());
+        }
+        if (!remoteServerConfiguration.getUrl().startsWith("https://")) {
+            log.warn("non-HTTPS connection to " + serverName);
+        }
+
+        WebResource resource = restClient.resource(remoteServerConfiguration.getUrl()).path("ws/rest/v1/patient").queryParam("v", "full");
+        if(timeout!=null){
+            restClient.setReadTimeout(timeout);
+        }
+        resource.addFilter(new HTTPBasicAuthFilter(remoteServerConfiguration.getUsername(), remoteServerConfiguration.getPassword()));
+        return resource;
+    }
+
+    public List<RemotePatient> searchRemoteServer(String serverName, String id, Integer timeout) throws IOException {
+        RemoteServerConfiguration remoteServerConfiguration = remoteServers.get(serverName);
+        WebResource resource = setUpWebResource(serverName, remoteServerConfiguration, timeout);
+
+        resource = resource.queryParam("id", id);
+
+        return doPatientSearch(remoteServerConfiguration, resource);
+    }
+
+    private String formatDate(Date date) {
+        return new SimpleDateFormat(FULL_DATE_FORMAT).format(date);
+    }
+
+    private void copyDateProperty(Object ontoBean, JsonNode fromJson, String field) {
+        String asText;
+        try {
+            asText = fromJson.get(field).getTextValue();
+        } catch (Exception ex) {
+            // skip fields not contained in the json
+            return;
+        }
+        if (StringUtils.isBlank(asText)) {
+            return;
+        }
+        Date date = parseDate(asText);
+        try {
+            PropertyUtils.setProperty(ontoBean, field, date);
+        } catch (Exception ex) {
+            throw new IllegalArgumentException(ex);
+        }
+    }
+
+    private void copyStringProperty(Object ontoBean, JsonNode fromJson, String field) {
+        String asText;
+        try {
+            asText = fromJson.get(field).getTextValue();
+        } catch (Exception ex) {
+            // skip fields not contained in the json
+            return;
+        }
+        try {
+            PropertyUtils.setProperty(ontoBean, field, asText);
+        } catch (Exception ex) {
+            throw new IllegalArgumentException(ex);
+        }
+    }
+
+    private Date parseDate(String date) {
+        try {
+            DateFormat dateFormat = new SimpleDateFormat(FULL_DATE_FORMAT);
+            return dateFormat.parse(date);
+        } catch (ParseException e) {
+            throw new IllegalArgumentException("Badly formatted date: " + date);
+        }
+    }
+
+}
```

Comment:
Assuming this is just moved wholesale from ImportFromWS, if there's anything in particular you think I should look at here, LMK...